smack-websocket/src/main/java/org/jivesoftware/smack/websocket/rce/RemoteWebsocketConnectionEndpoints.java

```diff
@@ -0,0 +1,54 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.websocket.rce;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+
+import org.jivesoftware.smack.c2s.internal.ModularXmppClientToServerConnectionInternal;
+import org.jivesoftware.smack.util.rce.RemoteConnectionEndpoint;
+import org.jivesoftware.smack.util.rce.RemoteConnectionEndpointLookupFailure;
+import org.jivesoftware.smack.websocket.implementations.OkHttpWebsocketImpl;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+
+public final class RemoteWebsocketConnectionEndpoints {
+
+    public static Result<WebsocketRemoteConnectionEndpoint> lookup(ModularXmppClientToServerConnectionInternal connectionInternal) {
+        List<RemoteConnectionEndpointLookupFailure> lookupFailures = null;
+
+        List<WebsocketRemoteConnectionEndpoint> discoveredRemoteConnectionEndpoints = null;
+        try {
+            discoveredRemoteConnectionEndpoints = OkHttpWebsocketImpl.getResultForRequest(connectionInternal.connection.getXMPPServiceDomain());
```

Comment:
Why is this method named `getResultForRequest`? Why is it in `OkHttpWeboscketImpl`, when it actually only uses your `HttpLookupMethod` API?